File: 1920-build-array-from-permutation/1920-build-array-from-permutation.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var buildArray = function(nums) {
    let resultArr = [];
    for(let i=0; i<nums.length; i++){
        resultArr.push(nums[nums[i]]);
    }
    return resultArr;
};

File: 1920-build-array-from-permutation/1920-build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        List<Integer> list = new ArrayList<Integer>();
        for(int i=0; i<nums.length; i++){
            list.add(nums[nums[i]]);
        }
        for(int i=0; i<list.size(); i++){
            nums[i] = list.get(i);
        }
        return nums;
    }
}